tests/flexmock_test.py

```diff
@@ -880,16 +920,31 @@ def __iter__(self):
         assert_equal(True, Foo.__iter__ == old, "%s != %s" % (Foo.__iter__, old))
 
     def test_flexmock_should_mock_private_methods_with_leading_underscores(self):
-        class _Foo:
-            def __stuff(self):
+        class ClassWithPrivateMethods:
+            def __private_instance_method(self):
                 pass
 
-            def public_method(self):
-                return self.__stuff()
+            @classmethod
+            def __private_class_method(cls):
+                # pylint: disable=unused-private-member
+                # pylint bug https://github.com/PyCQA/pylint/issues/4681
+                pass
 
-        foo = _Foo()
-        flexmock(foo).should_receive("__stuff").and_return("bar")
-        assert_equal("bar", foo.public_method())
+            def instance_method(self):
+                return self.__private_instance_method()
+
+            @classmethod
+            def class_method(cls):
+                return cls.__private_class_method()
+
+        # Instance
+        instance = ClassWithPrivateMethods()
```

Comment:
It would be interesting to see if it still works on a derived class, I'm expecting it won't, and thus it should be documented

Reply:
Can you give me some example so I can make a test case for that?

Reply:
```python
class DerivedClass(ClassWithPrivateMethods):
    pass

instance = DerivedClass()
flexmock(instance).should_receive("__private_instance_method").and_return("bar")
assert_equal("bar", instance.instance_method())  # if I'm not mistaken, this would fail, but I'm curious
```

Reply:
Right, it doesn't work:
```
flexmock.exceptions.FlexmockError: <tests.flexmock_test.RegularClass.test_flexmock_should_mock_private_methods_with_leading_underscores.<locals>.DerivedClass object at 0x7fda5ddabc10> does not have attribute _DerivedClass__private_instance_method
```
Should we create an issue about this? I think we could make it work with some magic.

Reply:
I think it should be expected that we would need to mock in the parent class, because that's how private methods work in Python. Now 7h after making that comment, I would say it works exactly as it should, so, even if my original plan was that we should document it, I am not sure anymore ; people using private methods should know that's how it is.

If we really want to do something we could do like Rust: `Hint: did you mean to mock method __private_instance_method of class ClassWithProvateMethods?` :D

Reply:
Yeah, you are right. It's better to just mock the parent class.